### PR TITLE
Add LayerZero cross chain support

### PR DESCRIPTION
**Pull request by jaydenwindle**

### CODE REVIEW

File: packages/sdk/src/functions/crossChain.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    abi: ERC_6551_DEFAULT.IMPLEMENTATION.ABI,
+    functionName: 'execute',
+    args: [to, value, data, 0],
+  })
+
+  const txOptions = Options.newOptions().addExecutorLzReceiveOption(200000, 0)
+
+  const txOptionsHex = txOptions.toHex() as `0x${string}`
+
+  const quoteData: any = await publicClient.readContract({
```
Comment (bjfresh) [inline packages/sdk/src/functions/crossChain.ts:45]: Looks like we could type this something like 

```
type QuoteData = {
  nativeFee: bigint,
  lzTokenFee: bigint
};
```

?

File: packages/sdk/src/TokenboundClient.ts [hunk 2]
```diff
@@ -418,22 +431,45 @@ class TokenboundClient {
...
+      const {
+        to: crossChainTo,
+        value: crossChainValue,
+        data: crossChainData,
+      } = await encodeCrossChainCall({
+        publicClient: this.publicClient,
+        account,
+        to,
+        value,
+        data: data as `0x${string}`,
```
Comment (bjfresh) [inline packages/sdk/src/TokenboundClient.ts:455]: Looks like we should tighten up typing on `ExecuteCallParams` in `params.ts` like so:

```
  data: `0x${string}`
```

so we don't necessitate explicit casting elsewhere.

File: packages/sdk/src/functions/crossChain.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    address: lzExecutor,
+    abi: lzExecutorAbi,
+    functionName: 'quote',
+    args: [lzEid, account, destinationExecutionData, txOptionsHex],
+  })
+
+  const txValue = quoteData.at(0) || 0
```
Comment (bjfresh) [inline packages/sdk/src/functions/crossChain.ts:52]: That'd catch on the 2nd value here, which should be 0n

**Review by bjfresh — APPROVED** [review]
Looks good, just a couple of little type issues that we should tighten up before shipping.

We can also bump viem/anvil + viem 1.x versions to the latest

**Comment by jaydenwindle** [comment]
@bjfresh great feedback! Going to avoid a version bump in viemfor now, can be a separate PR

**Pull request merged**